maven-model-builder/src/main/java/org/apache/maven/model/building/DefaultModelBuilder.java

```diff
@@ -482,6 +500,68 @@ else if ( !parentIds.add( parentData.getId() ) )
         return resultModel;
     }
 
+    private Map<String, Activation> getInterpolatedActivations( Model rawModel,
+                                                                DefaultProfileActivationContext context,
+                                                                DefaultModelProblemCollector problems )
+    {
+        Map<String, Activation> interpolatedActivations = getProfileActivations( rawModel, true );
+        for ( Activation activation : interpolatedActivations.values() )
+        {
+            if ( activation.getFile() != null )
+            {
+                replaceWithInterpolatedValue( activation.getFile(), context, problems );
+            }
+        }
+        return interpolatedActivations;
+    }
+
+    private void replaceWithInterpolatedValue( ActivationFile activationFile, ProfileActivationContext context,
+                                               DefaultModelProblemCollector problems  )
+    {
+        String path;
+        boolean missing;
```

Comment:
This missing variable feels complex, in a cyclomatic sense. It might be simpler to remove it and put more logic inside the first if-else if blocks. What do you think?